vowpalwabbit/parse_dispatch_loop.h

```diff
@@ -5,19 +5,38 @@
 #pragma once
 
 #include <functional>
+#include <thread>
+#include <future>
+#include <mutex>  
+#include "io_to_queue.h"
 
 using dispatch_fptr = std::function<void(vw&, const v_array<example*>&)>;
+struct IO_State;
 
 inline void parse_dispatch(vw& all, dispatch_fptr dispatch)
 {
   v_array<example*> examples = v_init<example*>();
   size_t example_number = 0;  // for variable-size batch learning algorithms
 
+  std::queue<IO_Item> *io_lines = new std::queue<IO_Item>;
+  IO_State io_state(io_lines);
+
+  all.p->_io_state = io_state;
```

Comment:
Is there a way we can make this passing of state more explicit? Right now it is stashed in the parser object and then moved over to the other thread by passing a reference to the all object. Generally if we can constrain what the other thread has access to we can avoid misuse and bugs.